SS2-Project/Assets/Starstorm2/Modules/Characters/Survivors/Pyro.cs

```diff
@@ -29,18 +35,47 @@ public override void Initialize()
             FlamethrowerDamageType = ReserveDamageType();
             FireballDamageType = ReserveDamageType();
 
+            _pyroBody = AssetCollection.FindAsset<GameObject>("PyroBody");
+
             _hotFireVFX = AssetCollection.FindAsset<GameObject>("PyroHotFireVFX");
             _fireballExplosionVFX = AssetCollection.FindAsset<GameObject>("PyroFireballExplosionVFX");
             _bdPyroManiac = AssetCollection.FindAsset<BuffDef>("bdPyroManiac");
             _bdPyroJet = AssetCollection.FindAsset<BuffDef>("bdPyroJet");
+            _bdPyroJetHiddenBoost = AssetCollection.FindAsset<BuffDef>("bdPyroJetHidden");
+            _jetpackOverrideDef = AssetCollection.FindAsset<HeatSkillDef>("sdPyro3a");
 
-            On.RoR2.Projectile.ProjectileSingleTargetImpact.OnProjectileImpact += PSTI_OPI;
-
-            //GlobalEventManager.onServerDamageDealt += PyroDamageChecks;
             On.RoR2.HealthComponent.TakeDamage += HealthComponent_TakeDamage;
-            //On.RoR2.DotController.OnDotStackAddedServer += DotController_OnDotStackAddedServer;
+            On.RoR2.DotController.InflictDot_refInflictDotInfo += DotController_InflictDot_refInflictDotInfo;
+            On.RoR2.BodyCatalog.SetBodyPrefabs += BodyCatalog_SetBodyPrefabs;
```

Comment:
Instead of this, please subscribe to ``BodyCatalog.availability``, it will cause the method you give it to be called as soon as the catalog is ready to be used.

Beats having an unecesary hook